include/flatbuffers/flexbuffers.h

```diff
@@ -1496,7 +1506,11 @@ class Builder FLATBUFFERS_FINAL_CLASS {
     auto byte_width = Align(bit_width);
     Write<uint64_t>(len, byte_width);
     auto sloc = buf_.size();
-    WriteBytes(data, len + trailing);
+    if (data != nullptr) {
```

Comment:
Let's not have this kind of "magic" functionality hidden in the main function, but instead have a designated `CreateUninitializedBlob` function. Then we can also simplify by omitting all parameters except for `len`, and we can return a pointer rather than a `size_t` so we don't need `GetUnfinishedBuffer` anymore. And please document that the pointer should be written to before anything else is serialized, instead of just saying "footgun" :)

Reply:
Check, `CreateUninitialized...` makes sense, will do.

But why does the pointer need to be written to before anything else is serialized? If that's not required, then I can, for instance, create such an uninitialized blob, and much later write data into the flexbuffer (by getting the Blob from the root, getting it's data() ptr, and casting away const - totally safe I assure you), and it all seems to work...

Reply:
because it is a pointer to a `std::vector` buffer, which may invalidate that pointer on any `push_back` or resize.